### PR TITLE
add colormode and colorscheme toggles to storybook toolbar

### LINKED ISSUE #1433: Preview components in different themes
## Background

Primer React components can be rendered in many [different themes](https://primer.style/primitives/colors#themes). See the [Theming](https://primer.style/react/theming) for more details.

| Light | Dark |
| --- | --- |
| CleanShot 2021-09-16 at 16 27 32 | CleanShot 2021-09-16 at 16 28 17 |

In fact, one of the criteria for a component to be [alpha](https://primer.style/contribute/component-lifecycle#alpha) is that "the component is compatible with color modes and can adapt to different themes."

## Problem

We don't have an easy way to preview components in different themes in documentation examples and Storybook.

## Tasks

- [ ] Implement a way to preview components in different themes in documentation examples
- [x] Implement a way to preview components in different themes in Storybook https://github.com/primer/react/pull/1436

### PR DESCRIPTION
**Pull request by langermank**
Adds a [custom decorator](https://storybook.js.org/docs/react/essentials/toolbars-and-globals#create-a-decorator) that wraps all stories with `ThemeProvider`

The decorator controls `colorMode` and `colorScheme` for stories using a new dropdown menu in the toolbar

Handles the storybook implementation for #1433 

To test locally, checkout `storybook-exploration` and run `npm run start:storybook`

### Open questions

- Should we find a way to combine `dayScheme` and `nightScheme`? Without much experience with schemes, do they cancel each other out when both set?
- Are these menu icons clear enough? The [options](https://5a375b97f4b14f0020b0cda3-wbeulgbetj.chromatic.com/?path=/story/basics-icon--labels) are limited, unfortunately (they don't have a moon or sun) 🤷 

### Screenshots

Screen Shot 2021-09-17 at 10 53 38 AM

colormodes-storybook

### CODE REVIEW

File: .storybook/preview.js [hunk 1]
```diff
@@ -1,4 +1,63 @@
...
+      showName: true
+    }
+  },
+  dayScheme: {
+    name: 'Day color scheme',
+    description: 'Day color scheme',
+    defaultValue: 'light',
+    toolbar: {
+      icon: 'circlehollow',
+      items: ['light', 'dark', 'dark_dimmed'],
```
Comment (colebemis) [inline .storybook/preview.js:31]: We can avoid hard-coding these names here if we do something like this:

```js
import {theme} from '../src'

export const globalTypes = {
  ...
  dayScheme: {
    toolbar: {
      items: Object.keys(theme.colorSchemes)
    }
  }
}
```

File: .storybook/preview.js [hunk 2]
```diff
@@ -1,4 +1,63 @@
...
+    color: ${themeGet('colors.text.primary')}
+  }
+`
+export const globalTypes = {
+  colorMode: {
+    name: 'Color mode',
+    description: 'Color mode (day, night, auto)',
+    defaultValue: 'day',
+    toolbar: {
+      icon: 'paintbrush',
```
Comment (colebemis) [inline .storybook/preview.js:19]: Does storybook have a black and white circle icon something like this: 🌗?

File: .storybook/preview.js [hunk 3]
```diff
@@ -1,4 +1,63 @@
...
+    defaultValue: 'dark',
+    toolbar: {
+      icon: 'circle',
+      items: ['light', 'dark', 'dark_dimmed'],
+      showName: true
+    }
+  }
+}
+
+// context.globals.X references items in globalTypes
```
Comment (colebemis) [inline .storybook/preview.js:47]: This is a helpful comment. Thanks for including this!

File: .storybook/preview.js [hunk 4]
```diff
@@ -1,4 +1,63 @@
 import {addons} from '@storybook/addons'
+import {ThemeProvider} from '../src/ThemeProvider'
+import {themeGet} from '../src'
```
Comment (colebemis) [inline .storybook/preview.js:3]: ThemeProvider is also exported from `src/index.ts` so we can combine these imports:

```suggestion
import {ThemeProvider, themeGet} from '../src'
```

File: .storybook/preview.js [hunk 5]
```diff
@@ -1,4 +1,63 @@
...
+import {createGlobalStyle} from 'styled-components'
+
+// set global theme styles for each story
+const GlobalStyle = createGlobalStyle`
+  body {
+    background-color: ${themeGet('colors.bg.primary')};
+    color: ${themeGet('colors.text.primary')}
```
Comment (colebemis) [inline .storybook/preview.js:10]:
```suggestion
    color: ${themeGet('colors.text.primary')};
```

File: .storybook/preview.js (hunk 2 above)
Comment (colebemis) [reply .storybook/preview.js:19]: Looks like they don't 😞 I

 think a `paintbrush` should be fine 👍

**Comment by colebemis** [comment]
This is amazing, @langermank! 💖 Thank you for doing this.

These controls will make it much easier for us to build themeable components!

**Comment by colebemis** [comment]
> Should we find a way to combine dayScheme and nightScheme? Without much experience with schemes, do they cancel each other out when both set?

`dayScheme` is only active in `day` mode and `nightScheme` is only active in `night` mode. They don't conflict with each other when they're both set

File: .storybook/preview.js (hunk 2 above)
Comment (langermank) [reply .storybook/preview.js:19]: I'm sure there's a way to pass custom icons to the toolbar, but I couldn't find much after a quick search. Someday!

File: .storybook/preview.js [hunk 6]
```diff
@@ -1,4 +1,62 @@
...
+`
+export const globalTypes = {
+  colorMode: {
+    name: 'Color mode',
+    description: 'Color mode (day, night, auto)',
+    defaultValue: 'day',
+    toolbar: {
+      icon: 'paintbrush',
+      // array of colorMode items
+      items: Object.keys(theme.colorSchemes),
```
Comment (colebemis) [inline .storybook/preview.js:20]: This should stay hard-coded 😅 

"color modes" and "color schemes" are _different_ (but related) ideas in Primer React.

```suggestion
      items: ['day', 'night', 'auto'],
```

**Review by colebemis — APPROVED** [review]
Woooo! 🎉

**Pull request merged**